### PR TITLE
ISSUE-822 Webadmin task for spam cleanup

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

**Comment by Arsnael** [comment]
Missing (yes sorry doing multiple things at the same time^^):

- additional details (successful users, failed users, number of successful message, number of failed messages)?
- usersPerSecond -> messagePerSecond in RunningOptions, as caught by Tung on trash cleanup task, with a concurrency of 1 for users, and leveraging down the throttle on messages instead of users

File: tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupSpamService.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    private final UsersRepository usersRepository;
+    private final JmapSettingsRepository jmapSettingsRepository;
+    private final StoreMailboxManager storeMailboxManager;
+    private final SystemMailboxesProvider systemMailboxesProvider;
+    private final Clock clock;
+
+    @Inject
+    public CleanupSpamService(UsersRepository usersRepository, JmapSettingsRepository jmapSettingsRepository, StoreMailboxManager storeMailboxManager,
+                               SystemMailboxesProvider systemMailboxesProvider, Clock clock) {
```
Comment (quantranhong1999) [inline tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupSpamService.java:44]:
```suggestion
    public CleanupSpamService(UsersRepository usersRepository, JmapSettingsRepository jmapSettingsRepository, StoreMailboxManager storeMailboxManager,
                              SystemMailboxesProvider systemMailboxesProvider, Clock clock) {
```

File: tmail-backend/integration-tests/webadmin/webadmin-integration-tests-common/src/main/java/com/linagora/tmail/integration/CleanupIntegrationContract.java [hunk 2]
```diff
@@ -73,4 +73,49 @@ void cleanupTrashTaskShouldFailWhenQueryParaValueIsInvalid() {
...
+            .queryParam("task", "CleanupSpam")
+            .post()
+        .then()
+            .statusCode(201)
+            .body("taskId", notNullValue());
+    }
+
+    @Test
+    void cleanupSpamTaskShouldWork() {
+        String taskId = given()
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/webadmin/webadmin-integration-tests-common/src/main/java/com/linagora/tmail/integration/CleanupIntegrationContract.java:89]: Would be better if we could put a message in Spam and check we delete it IMO.

**Review by quantranhong1999 — COMMENTED** [review]
Looks good to me so far

**Comment by vttranlina** [comment]
Can we re-use code from CleanupTrash?
I see almost code of Trash and Spam is similar, 
How about refactoring `CleanupTrashService` to `CleanupMailboxService`?

**Comment by Arsnael** [comment]
Yes yes, that's what I did but based on yesterday's morning code of TrashCleanup. There is definitely space for refactoring :)

**Comment by Arsnael** [comment]
I'm waiting for https://github.com/linagora/tmail-backend/pull/829 to be merged before rebasing again. It's hellish each time.

File: docs/modules/ROOT/pages/tmail-backend/webadmin.adoc [hunk 3]
```diff
@@ -638,3 +638,39 @@ Valid status includes:
...
+
+Return codes:
+
+ - `201` Success. Corresponding task id is returned.
+ - `400` Error in the request. Details can be found in the reported error.
+
+=== Clean Spam
+
+....
+curl -XGET http://ip:port/mailboxes?task=CleanupSpam&usersPerSecond={usersPerSecondValue}
```
Comment (quantranhong1999) [inline docs/modules/ROOT/pages/tmail-backend/webadmin.adoc:664]:
```suggestion
curl -XPOST http://ip:port/mailboxes?task=CleanupSpam&usersPerSecond={usersPerSecondValue}
```

**Review by vttranlina — APPROVED** [review]
Looks good!

// I thought about using single Task, DTO, Details classes for both Spam, Trash. 
But it looks complexity

**Comment by Arsnael** [comment]
> // I thought about using single Task, DTO, Details classes for both Spam, Trash. But it looks complexity

Tried that at first, then it became an absolute hell of Guice multiple bindings issues, so I decided to stop :)

**Review by quantranhong1999 — APPROVED** [review]

**Comment by vttranlina** [comment]
Need resolve conflict

**Comment by Arsnael** [comment]
> Need resolve conflict

Done

**Pull request merged**